classes/plugins/ThemePlugin.inc.php

```diff
@@ -877,6 +881,146 @@ public function isColourDark($color, $limit = 130)
+            'pkpUsageStatsData',
+            $script_data,
+            [
+                'inline' => true,
+                'contexts' => $this->getSubmissionViewContext(),
+            ]
+        );
+    }
+
+    /**
+     * Load JavaScript assets for usage statistics display and pass data to the scripts
+     */
+    protected function loadChartJavascript(): void
+    {
+        $request = Application::get()->getRequest();
+        $templateMgr = TemplateManager::getManager($request);
+
+        // Register Chart.js on the frontend article view
+        $templateMgr->addJavaScript(
+            'chartJS',
+            'https://cdnjs.cloudflare.com/ajax/libs/Chart.js/2.0.1/Chart.js',
+            [
+                'contexts' => $this->getSubmissionViewContext(),
+            ]
+        );
+
+        // Add locale and configuration data
+        $chartType = $this->getOption('displayStats');
+        $script_data = 'var pkpUsageStats = pkpUsageStats || {};';
+        $script_data .= 'pkpUsageStats.locale = pkpUsageStats.locale || {};';
+        $script_data .= 'pkpUsageStats.locale.months = ' . json_encode(explode(' ', __('plugins.themes.default.displayStats.monthInitials'))) . ';';
+        $script_data .= 'pkpUsageStats.config = pkpUsageStats.config || {};';
+        $script_data .= 'pkpUsageStats.config.chartType = ' . json_encode($chartType) . ';';
+
+        $templateMgr->addJavaScript(
+            'pkpUsageStatsConfig',
+            $script_data,
+            [
+                'inline' => true,
+                'contexts' => $this->getSubmissionViewContext(),
+            ]
+        );
+
+        // Register the JS which initializes the chart
+        $templateMgr->addJavaScript(
+            'usageStatsFrontend',
+            $request->getBaseUrl() . '/lib/pkp/js/usage-stats-chart.js',
+            [
+                'contexts' => $this->getSubmissionViewContext(),
+            ]
+        );
+    }
+
+    /**
+     * Retrieve download metrics for the given submission
+     */
+    protected function getAllDownloadsStats(int $submissionId): array
+    {
+        $cache = CacheManager::getManager()->getCache('downloadStats', $submissionId, [$this, 'downloadStatsCacheMiss']);
+        if (time() - $cache->getCacheTime() > 60 * 60 * 24) {
+            // Cache is older than one day, erase it.
+            $cache->flush();
+        }
+        $statsByMonth = [];
+        $totalDownloads = 0;
+        $data = $cache->get($submissionId);
+        foreach ($data as $monthlyDownloadStats) {
+            [$year, $month] = explode('-', $monthlyDownloadStats['date']);
+            $month = ltrim($month, '0');
+            $statsByMonth[$year][$month] = $monthlyDownloadStats['value'];
+            $totalDownloads += $monthlyDownloadStats['value'];
+        }
+        return [
+            'data' => $statsByMonth,
+            'label' => __('common.allDownloads'),
+            'color' => $this->getUsageStatsDispalyColor(REALLY_BIG_NUMBER),
```

Comment:
Typo: `Display`

Reply:
(Also, why is `REALLY_BIG_NUMBER` being used here?)

Reply:
To be hones, I do not know -- I've kept it from UsageStatsPlugin, s. https://github.com/pkp/usageStats/blob/main/UsageStatsPlugin.inc.php#L870.

Reply:
OK, let's leave it as is for now.